computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmComputationManager.java

```diff
@@ -215,26 +217,47 @@ public OutputStream newCommonFile(String fileName) throws IOException {
                 LOGGER.error(e.toString(), e);
                 f.completeExceptionally(e);
             }
-            taskStore.remove(f);
+            if (!f.isCancelled()) {
+                taskStore.clearBy(f);
+            }
         });
         return f;
     }
 
-    static class Mycf<R> extends CompletableFuture<R> {
+    static class SlurmCompletableFuture<R> extends CompletableFuture<R> {
 
-        Thread thread;
-        SlurmComputationManager mgr;
-        volatile boolean cancel = false;
+        private Thread thread;
+        private SlurmComputationManager mgr;
+        private volatile boolean cancel = false;
+        private volatile boolean cancelledBySlurm = false;
 
-        Mycf(SlurmComputationManager manager) {
-            mgr = manager;
+        private SlurmException exception;
+
+        SlurmCompletableFuture(SlurmComputationManager manager) {
+            mgr = Objects.requireNonNull(manager);
         }
 
         @Override
         public boolean isCancelled() {
             return cancel;
         }
 
+        // called by monitors, it differs from cancel() by client-side
+        boolean cancelBySlurm(SlurmException exception) {
```

Comment:
I would propose to replace this with a plain call to `completeExceptionally(exception)`.

Reply:
As above: 
let's postpone this possible refactoring to after we have some feedback on the PR linked above.